src/main/java/edu/byu/cs/controller/SubmissionController.java

```diff
@@ -558,24 +565,50 @@ public static void approveSubmission(
             LOGGER.warn("Approving submissions did not affect any submissions. Something probably went wrong.");
         }
 
-        // Determine approvedScore
-        if (approvedScore == null) {
-            float bestScoreForPhase = submissionDao.getBestScoreForPhase(studentNetId, phase);
-            if (bestScoreForPhase < 0.0f) {
-                throw new RuntimeException("Cannot determine best score for phase without any submissions in the phase.");
+        if (submissionToUse == null) {
+            submissionToUse = submissionDao.getBestSubmissionForPhase(studentNetId, phase);
+            if (submissionToUse == null) {
+                throw new RuntimeException("No submission was provided nor found for phase " + phase + " with user " + studentNetId);
             }
-            approvedScore = SubmissionHelper.prepareModifiedScore(bestScoreForPhase, scoreVerification);
         }
-        if (approvedScore <= 0.0f) {
-            throw new RuntimeException("Cannot set grade without a positive approvedScore!");
+        if (approvedScore == null) {
+            approvedScore = submissionToUse.score() * (1 - (penaltyPct / 100));
         }
+        //approvedScore = SubmissionHelper.prepareModifiedScore(submissionToUse.score(), scoreVerification);
 
+        //TODO: Make it send the actual score with git and late penalty
         // Update grade-book
-        CanvasIntegration canvasIntegration = CanvasService.getCanvasIntegration();
-        // FIXME: Store `approvedScore` in the Grade-book
-        // canvasIntegration.submitGrade(studentNetId, approvedScore, );
-        if (true) {
-            throw new RuntimeException("ApproveSubmission not implemented!"); // TODO: Finish implementing method
+        float scoreDifference = originalScore - approvedScore;
+        RubricConfig rubricConfig = DaoService.getRubricConfigDao().getRubricConfig(phase);
+        Rubric oldRubric = submissionToUse.rubric();
+        Rubric rubicToUse = new Rubric(
+                oldRubric.passoffTests(),
+                oldRubric.unitTests(),
+                oldRubric.quality(),
+                new Rubric.RubricItem(
+                        "Git Commits",
+                        new Rubric.Results(
+                            null,
+                            scoreDifference,
+                            0,
+                            null,
+                            null),
+                        "Regularly commit to your Github"),
+                submissionToUse.passed(),
+                "Submission initially blocked due to low commits. Submission approved by admin " + approverNetId);
+
+
+        int canvasUserId = DaoService.getUserDao().getUser(studentNetId).canvasUserId();
+        int assignmentNum = PhaseUtils.getPhaseAssignmentNumber(phase);
+
+        // TODO: get assessment to hold git commit penalty
```

Comment:
Is this `TODO` comment still valid, or has it already been handled?

Reply:
I'm not sure what that is for, why would the git commit hold the penalty? My understanding is that it was standard. I was going to make the git commit penalty amount something that is stored in the config table when I get to that in the branch for #324